src/huggingface_hub/hub_mixin.py

```diff
@@ -267,56 +254,36 @@ def push_to_hub(
             The url of the commit of your model in the given repository.
         """
 
-        if repo_path_or_name is None and repo_url is None:
-            raise ValueError(
-                "You need to specify a `repo_path_or_name` or a `repo_url`."
-            )
+        token, _ = hf_api._validate_or_retrieve_token(token)
+        api = HfApi(endpoint=api_endpoint)
 
-        if use_auth_token is None and repo_url is None:
-            token = HfFolder.get_token()
-            if token is None:
-                raise ValueError(
-                    "You must login to the Hugging Face hub on this computer by typing `huggingface-cli login` and "
-                    "entering your credentials to use `use_auth_token=True`. Alternatively, you can pass your own "
-                    "token as the `use_auth_token` argument."
-                )
-        elif isinstance(use_auth_token, str):
-            token = use_auth_token
-        else:
-            token = None
-
-        if repo_path_or_name is None:
-            repo_path_or_name = repo_url.split("/")[-1]
-
-        # If no URL is passed and there's no path to a directory containing files, create a repo
-        if repo_url is None and not os.path.exists(repo_path_or_name):
-            repo_id = Path(repo_path_or_name).name
-            if organization:
-                repo_id = f"{organization}/{repo_id}"
-            repo_url = HfApi(endpoint=api_endpoint).create_repo(
-                repo_id=repo_id,
-                token=token,
-                private=private,
-                repo_type=None,
-                exist_ok=True,
-            )
-
-        repo = Repository(
-            repo_path_or_name,
-            clone_from=repo_url,
-            use_auth_token=use_auth_token,
-            git_user=git_user,
-            git_email=git_email,
+        api.create_repo(
+            repo_id=repo_id,
+            token=token,
+            private=private,
+            repo_type=None,
+            exist_ok=True,
         )
-        repo.git_pull(rebase=True)
 
         # Save the files in the cloned repo
-        self.save_pretrained(repo_path_or_name, config=config)
-
-        # Commit and push!
-        repo.git_add()
-        repo.git_commit(commit_message)
-        return repo.git_push()
+        with tempfile.TemporaryDirectory() as tmp:
+            saved_path = Path(tmp) / repo_id
+            self.save_pretrained(saved_path, config=config)
+
+            for path, currentDirectory, files in os.walk(saved_path):
+                for filename in files:
+                    file = os.path.join(path, filename)
+                    common_prefix = os.path.commonprefix([saved_path, file])
+                    relative_path = os.path.relpath(file, common_prefix)
+
+                    api.upload_file(
```

Comment:
In the docs it says `upload_file` should be used when the file size is less than 10MB: https://huggingface.co/docs/hub/adding-a-model

Can this new implementation handle weights files having 100-200MB size?

Reply:
This implementation can handle weights files up to 30GB!

Reply:
You're correct that the documentation should likely be updated to reflect that.

Reply:
Will be discussed in a separate issue https://github.com/huggingface/huggingface_hub/issues/995.